### ISSUE TITLE
Handling out of order responses from Promises

### ISSUE THREAD

**Issue by yelouafi** [paldepind/flyd#37]
``` javascript
function updateStreamValue(s, n) {
  if (n !== undefined && n !== null && isFunction(n.then)) {
    n.then(s); // here??
    return;
  }
  ...
}
```

If the promises are resolved in an arbitrary order (eg. the second promise resolve before the first) then the stream wll be updated with an outdated value

**Comment by iofjuupasli** [comment]
https://github.com/paldepind/flyd-switchlatest is what you want probably

**Comment by yelouafi** [comment]
Yes, that's the way to do in other libs like bacon.js. If it's this is the desired behavior it' be preferable to mention it in the docs.

**Comment by paldepind** [comment]
I'm not particularly happy with this behavior.

**Comment by paldepind** [comment]
The tricky thing is that sometimes order matters and in some cases it don't. And when it doesn't matter you don't want slow promises to block results from later promises.

**Comment by yelouafi** [comment]
IMO the default behavior should be preserving the order. The other case can be handled by something like

```javascript
stream$.flatmap( v => flyd.fromPromise( $.ajax(...) )
```

**Comment by paldepind** [comment]
I agree.

**Comment by corps** [comment]
I believe the default behavior should do /the least special thing/.  That's because the "I expect responses to preserve ordering" problem can be solved in at least two different ways:

1.  A response that comes later than a more recent response is simply dropped and not pushed down the stream (think making multiple update requests, where only the most recent update request is valid).  *This drops responses that arrive later than their ordering in the list*.

2.  All responses are pushed into the stream, but each response is dependent on the previous request completing.  *This drops responses that arrive sooner than their ordering in the list, if the dependent responses error or never arrive*.

I don't think there actually exists a single "intuitive" approach to this async problem, and it largely depends on what you're solving.  The current default behavior is "dumb" and adds no special logic, leaving that as a problem to be solved with more specialized stream utilities.

You should not turn your brain off during async handling or trust in a default behavior.  Always assume that things both come out of order and are dropped from responding at all.

**Comment by paldepind** [comment]
This issue is resolved by #168. The inbuilt promise handling is being removed. The new way to handle promises is documented [here](https://github.com/paldepind/flyd#promises) and it makes it possible to ensure the order of resolved values. Big thanks to @nordfjord.
